runtime/ops/command.rs

```diff
@@ -0,0 +1,359 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use super::io::ChildStderrResource;
+use super::io::ChildStdinResource;
+use super::io::ChildStdoutResource;
+use crate::permissions::Permissions;
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::op_async;
+use deno_core::op_sync;
+use deno_core::AsyncRefCell;
+use deno_core::Extension;
+use deno_core::OpState;
+use deno_core::RcRef;
+use deno_core::Resource;
+use deno_core::ResourceId;
+use deno_core::ZeroCopyBuf;
+use serde::Deserialize;
+use serde::Serialize;
+use std::borrow::Cow;
+use std::cell::RefCell;
+use std::process::ExitStatus;
+use std::rc::Rc;
+use tokio::process::Command;
+
+#[cfg(unix)]
+use std::os::unix::prelude::ExitStatusExt;
+
+pub fn init() -> Extension {
+  Extension::builder()
+    .ops(vec![
+      ("op_create_command", op_sync(op_create_command)),
+      ("op_command_spawn", op_sync(op_command_spawn)),
+      ("op_command_status", op_async(op_command_status)),
+      ("op_command_output", op_async(op_command_output)),
+      ("op_command_child_wait", op_async(op_command_child_wait)),
+      ("op_command_child_output", op_async(op_command_child_output)),
+      ("op_command_child_status", op_sync(op_command_child_status)),
+    ])
+    .build()
+}
+
+struct ChildResource(AsyncRefCell<tokio::process::Child>);
+
+impl Resource for ChildResource {
+  fn name(&self) -> Cow<str> {
+    "child".into()
+  }
+}
+
+struct CommandResource(tokio::process::Command);
+
+impl Resource for CommandResource {
+  fn name(&self) -> Cow<str> {
+    "command".into()
+  }
+}
+
+fn subprocess_stdio_map(s: &str) -> Result<std::process::Stdio, AnyError> {
+  match s {
+    "inherit" => Ok(std::process::Stdio::inherit()),
+    "piped" => Ok(std::process::Stdio::piped()),
+    "null" => Ok(std::process::Stdio::null()),
+    _ => Err(type_error("Invalid resource for stdio")),
+  }
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandIoArgs {
+  stdin: Option<String>,
+  stdout: Option<String>,
+  stderr: Option<String>,
```

Comment:
I would strongly recommend using an enum instead of a string:
```rust
#[derive(Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum Stdio {
  Inherit,
  Piped,
  Null,
}
```
And then:
```suggestion
  stdin: Option<Stdio>,
  stdout: Option<Stdio>,
  stderr: Option<Stdio>,
```